### DISCUSSION TITLE
Dynamic Routes in NextJS 10 don't return 404s when they don't exist?

### DISCUSSION THREAD

**Discussion by pkellner** [vercel/vercel#5626] [Help]
Running the example in the repo:

https://github.com/vercel/next.js/tree/canary/examples/dynamic-routing

I notice that 

But, this: http://localhost:3000/post/firstBADROUTE

returns a status code 200.

Any way to make it return a 404?  Guessing it may be the nature of the beast with static pages.

**Comment by dipakparmar** [comment]
> Running the example in the repo:
> 
> https://github.com/vercel/next.js/tree/canary/examples/dynamic-routing
> 
> I notice that
> 
> But, this: http://localhost:3000/post/firstBADROUTE
> 
> returns a status code 200.
> 
> Any way to make it return a 404? Guessing it may be the nature of the beast with static pages.

Hi @pkellner you can just remove [index.js](https://raw.githubusercontent.com/vercel/next.js/canary/examples/dynamic-routing/pages/post/%5Bid%5D/index.js) file, as index.js file is the base page for the post/[id] route it returns 200 with index.js page!

**Reply by pkellner** [reply]
Here is what I was looking for. https://github.com/vercel/next.js/discussions/10960#discussioncomment-116705
